kafka-ui-react-app/src/redux/reducers/topicMessages/topicMessagesSlice.ts

```diff
@@ -0,0 +1,92 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import { TopicMessagesState, ClusterName, TopicName } from 'redux/interfaces';
+import { TopicMessage, Configuration, MessagesApi } from 'generated-sources';
+import { BASE_PARAMS } from 'lib/constants';
+import { getResponse } from 'lib/errorHandling';
+import { showSuccessAlert } from 'redux/reducers/alerts/alertsSlice';
+
+const apiClientConf = new Configuration(BASE_PARAMS);
+export const messagesApiClient = new MessagesApi(apiClientConf);
+
+export const clearTopicMessages = createAsyncThunk<
+  undefined,
+  { clusterName: ClusterName; topicName: TopicName; partitions?: number[] }
+>(
+  'topicMessages/clearTopicMessages',
+  async (
+    { clusterName, topicName, partitions },
+    { rejectWithValue, dispatch }
+  ) => {
+    try {
+      await messagesApiClient.deleteTopicMessages({
+        clusterName,
+        topicName,
+        partitions,
+      });
+
+      dispatch(
+        showSuccessAlert({
+          id: `message-${topicName}-${clusterName}-${partitions}`,
+          message: 'Messages successfully cleared!',
+        })
+      );
+
+      return undefined;
+    } catch (err) {
+      return rejectWithValue(await getResponse(err as Response));
+    }
+  }
+);
+
+export const initialState: TopicMessagesState = {
+  messages: [],
+  meta: {
+    bytesConsumed: 0,
+    elapsedMs: 0,
+    messagesConsumed: 0,
+    isCancelled: false,
+  },
+  isFetching: false,
+};
+
+export const topicMessagesSlice = createSlice({
+  name: 'topicMessages',
+  initialState,
+  reducers: {
+    addTopicMessage: (state, action) => {
+      const messages: TopicMessage[] = action.payload.prepend
+        ? [action.payload.message, ...state.messages]
+        : [...state.messages, action.payload.message];
+
+      return {
+        ...state,
+        messages,
+      };
+    },
+    resetTopicMessages: () => initialState,
+    updateTopicMessagesPhase: (state, action) => {
+      state.phase = action.payload;
+    },
+    updateTopicMessagesMeta: (state, action) => {
+      state.meta = action.payload;
+    },
+    setTopicMessagesFetchingStatus: (state, action) => {
+      state.isFetching = action.payload;
```

Comment:
a question didn't we have a loader slicer that did this for us , where we added fetch and other state ? so we did not have to keep a loader state in a specific slice logic

Reply:
I got you but the problem here is that we don't have separate action for fetching topicMessages as you can see that's why we have to keep this logic for this case in slice